pkg/auth/providers/azure/azure_provider.go

```diff
@@ -400,6 +415,28 @@ func formAzureRedirectURL(config map[string]interface{}) string {
 	)
 }
 
+func extractAnnotations(config map[string]interface{}) map[string]string {
+	annotations := make(map[string]string)
+	metadata, ok := config["metadata"].(map[string]interface{})
+	if !ok {
+		logrus.Infof("Failed to decode the %q field of the auth config.", "metadata")
+		return annotations
+	}
+	rawAnnotations, ok := metadata["annotations"].(map[string]interface{})
+	if !ok {
+		logrus.Infof("Failed to decode the %q field of the auth config.", "annotations")
```

Comment:
Nit:

Same comment here on the use of a format string when the only arg is a constant.

Reply:
Adjusted!